cdap-master/src/main/java/co/cask/cdap/data/tools/Main.java

```diff
@@ -197,6 +203,54 @@ private void upgradeSystemDatasets(Injector injector) throws Exception {
     }
   }
 
+  private void migrateMetricsData(Injector injector) throws Exception {
+    // find version to migrate
+    ProjectInfo.Version version = findMetricsTableVersion(injector.getInstance(CConfiguration.class),
+                                                  injector.getInstance(Configuration.class));
+    if (version != null) {
+      DatasetFramework framework = createRegisteredDatasetFramework(injector);
+      // migrate metrics data
+      DefaultMetricDatasetFactory.migrateMetricsData(injector.getInstance(CConfiguration.class), framework, version);
+    }
+  }
+
+  private ProjectInfo.Version findMetricsTableVersion(CConfiguration cConf, Configuration hConf) {
+    // Upgrade all datasets in system namespace
+    boolean metricsTable27Found = false;
+    boolean metricsTable26Found = false;
```

Comment:
also: should you do this in the metrics migrate tool? Or, in the Metrics version tool?